### ISSUE TITLE
Helm Chart | Need to auto-update the version number in the Charts read.me

### ISSUE THREAD

**Issue by TracyRagan** [ortelius/ortelius#558]
We need to auto-update the https://github.com/ortelius/ortelius-charts/blob/main/charts/ortelius/README.md each time a workflow executes. Right now we must do it manually. 

Need to pull the version number from the chart and add it to this section of the read.me. For example updating the 10.0.77 to 10.0.78: 

```![Version: 10.0.77](https://img.shields.io/badge/Version-10.0.77-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: v10.0.0](https://img.shields.io/badge/AppVersion-v10.0.0-informational?style=flat-square)```

In addition, we need append the same read.me with a change log. The change log can be derived from the git commit history.  Fo example, if we increment to 10.0.78, a new "Changes in" should be added.

```## Additional Information

This chart deploys all of the required secrets, services, and deployments on a [Kubernetes](https://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.

### Changes in 10.0.54 - 10.0.77
Updated all base images to remove vulnerabilities. Used cgr.dev/chainguard/python, cgr.dev/chainguard/jdk:latest, and python:3.10-alpine, eclipse-temurin:8-jdk-alpine base images. The cgr.dev ones work great if you do not have additional packages (openssl) to install otherwise alpine is needed.

cgr.dev is based on wolfi. Google distroless has vulnerabilities in the base image so they should not be used.


### Changes in 10.0.53

* Added new Microservice for ScoreCard
* Updated ingress for running local Kind Cluster
* Removed dependency on external jwt keys ```

**Comment by sbtaylor15** [comment]
Change is needed in this workflow - https://github.com/ortelius/ortelius-charts/blob/main/.github/workflows/generate-main-chart.yml

**Comment by sbtaylor15** [comment]
`sudo go install github.com/mikefarah/yq/v4@latest` will install yq that can be used to pull Chart version from the Chart.yaml

**Comment by sbtaylor15** [comment]
Another way to go is to update the main.js to read in the README.md, update it and write out a new README.md

**Comment by sachajw** [comment]
I am thinking some extra logic [here](https://github.com/ortelius/ortelius-charts/blob/main/main.js) in Ortelius Charts could do the trick.

**Comment by sachajw** [comment]
Let's also consider that the simplest might be as easy as a hook https://github.com/norwoodj/helm-docs
I have used this hook many times and can confirm it works very well.
![image](https://user-images.githubusercontent.com/32266273/220337776-ba4ea217-bafb-403b-9a09-15b26fa5c120.png)

**Comment by sachajw** [comment]
Tests are looking good 
![image](https://user-images.githubusercontent.com/32266273/222729480-84451c84-56f7-4ee1-ad4d-ba76adcc739e.png)
![image](https://user-images.githubusercontent.com/32266273/222729536-4735c27d-6f4c-4811-9d75-f60a23f10f89.png)

Next step is to combine all the readmes into one.

**Comment by sachajw** [comment]
Example Helm doc with chart dependencies https://github.com/norwoodj/helm-docs/blob/master/example-charts/umbrella/README.md

**Comment by sachajw** [comment]
### Steps
1. [`Helm-Docs`](https://github.com/norwoodj/helm-docs) generates the readmes with a hook
2. The Python script merges all the readmes into one which is called [`MERGED-README.md`](https://github.com/sachajw/ortelius-charts-testing/blob/main/MERGED_README.md) at the moment
3. Use GitHub actions to automate the process

readme-merge.py
```
import os

# Define the root directory where README files are located
root_directory = "./"

# Define a function that recursively finds all README files in subdirectories
def find_readme_files(directory):
    readme_files = []
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file == "README.md":
                readme_files.append(os.path.join(root, file))
    return readme_files

# Call the function to find all README files
readme_files = find_readme_files(root_directory)

# Define a function to merge multiple README files into one file
def merge_readme_files(readme_files):
    merged_content = ""
    for file in readme_files:
        with open(file, "r") as f:
            content = f.read()
            merged_content += content + "\n\n"
    return merged_content

# Call the function to merge all README files
merged_content = merge_readme_files(readme_files)

# Write the merged content to a new file called "MERGED_README.md" in the root directory
with open(os.path.join(root_directory, "MERGED_README.md"), "w") as f:
    f.write(merged_content)
```
```
name: Consolidate README files

on:
  push:
    branches: [ main ]

jobs:
  consolidate:
    runs-on: ubuntu-latest

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Consolidate README files
      run: python readme-merge.py

    - name: Commit changes
      uses: stefanzweifel/git-auto-commit-action@v4
      with:
        commit_message: Consolidate README files
```

**Comment by sachajw** [comment]
Can the Ortelius values.yaml in ortelius/values.yaml be the grand overload of all things Ortelius Helm Chart configuration?
Essentially, if you want to override any defaults of the Microservices, we can use that values.yaml
One ring to rule them all.
Then Helm-Docs just needs to create Helm README documentation from one file?
We would not need any extra steps to merge the child chart README's.
https://github.com/sachajw/ortelius-charts-testing/blob/main/ortelius/values.yaml

**Comment by sbtaylor15** [comment]
> Can the Ortelius values.yaml in ortelius/values.yaml be the grand overload of all things Ortelius Helm Chart configuration? Essentially, if you want to override any defaults of the Microservices, we can use that values.yaml One ring to rule them all. Then Helm-Docs just needs to create Helm README documentation from one file? We would not need any extra steps to merge the child chart README's. https://github.com/sachajw/ortelius-charts-testing/blob/main/ortelius/values.yaml

Yes, the values.yaml at the Parent chart will override the values in the child charts.  Just need to reference the child chart name when doing the override, i.e. `ms-nginx.replicaCnt: 3`.  Or you can override at the helm command line a `--set` parameters.

**Comment by sbtaylor15** [comment]
@sachajw I looked at the MERGED-README.md and its getting closer.  The values from the child charts are making it confusing to follow since the are just the derived values and not the ones you can override.  

We should use a new file `READEME.template` that contains "tag placholders" for the dynamic data to be substituted in, plus the static text.  See `from string import Template`. That module enables text substitution based on `${MY_VAR}` placeholder tag.

## Summary Section

![image](https://user-images.githubusercontent.com/14826572/223190313-30fb2bca-54b1-4105-acb9-7f83e6ceafba.png)

## Changes Section

This section should be derived from the git commit log.  Work will need to be done on the GH Actions to push across the commit message from the child chart update over to the ortelius-charts repo actions. 

![image](https://user-images.githubusercontent.com/14826572/223192949-76cb1082-e56b-4586-b959-8dc7a12053e6.png)

## Install Command Section

This section should be static text from the README.template since it does not change very often.

![image](https://user-images.githubusercontent.com/14826572/223191749-c8918773-6da6-490c-8ca2-d9556b3382fd.png)

## Parameters Section

This section should be static text from the README.template since it does not change very often.  And the default values are handled in the charts themselves. 

![image](https://user-images.githubusercontent.com/14826572/223192049-7d1847d6-d40a-4adc-8f79-1ff8fa79119a.png)

## Accessing the Ortelius UI After the Chart Install Section

This section should be static text from the README.template since it does not change very often.  

![image](https://user-images.githubusercontent.com/14826572/223192230-8dbf5104-e09b-42f9-aa78-adbc0709accb.png)

## Uninstalling Section

This section should be static text from the README.template since it does not change very often.  

![image](https://user-images.githubusercontent.com/14826572/223192500-f4fcec64-82cf-4035-b00f-8c449ac96533.png)

**Comment by sachajw** [comment]
@sbtaylor15 awesome find to solve this issue! ![GitHub release (latest SemVer)](https://img.shields.io/github/v/release/ortelius/ortelius-charts)

**Comment by sbtaylor15** [comment]
Fixed ortelius/ortelius-charts#11
